octobot_trading/personal_data/positions/position.py

```diff
@@ -707,16 +729,22 @@ def _update_side(self):
         Only relevant when account is using one way position mode
         """
         if self.symbol_contract.is_one_way_position_mode() or self.side is enums.PositionSide.UNKNOWN:
+            changed_side = False
             if self.quantity >= constants.ZERO:
                 if self.side is not enums.PositionSide.LONG:
                     self.side = enums.PositionSide.LONG
-                    self._reset_entry_price()
+                    changed_side = True
             elif self.quantity < constants.ZERO:
                 if self.side is not enums.PositionSide.SHORT:
                     self.side = enums.PositionSide.SHORT
-                    self._reset_entry_price()
+                    changed_side = True
             else:
                 self.side = enums.PositionSide.UNKNOWN
+            if changed_side:
+                self._reset_entry_price()
+                self.exit_price = constants.ZERO
+                self.first_entry_time = self.exchange_manager.exchange.get_exchange_current_time()
+                logging.get_logger(self.get_logger_name()).info(f"Changed position side: now {self.side.name}")
```

Comment:
I think it should be clearer with a method to handle this event like `on_side_update`